static/js/things.js

```diff
@@ -300,6 +309,73 @@ const ThingsScreen = {
       });
     });
   },
+
+  /**
+   * Display an events list.
+   *
+   * @param {String} thingId The ID of the Thing to show.
+   */
+  showEvents: function(thingId) {
+    const opts = {
+      headers: {
+        Authorization: `Bearer ${API.jwt}`,
+        Accept: 'application/json',
+      },
+    };
+
+    // Fetch a thing from the server
+    fetch(`/things/${encodeURIComponent(thingId)}`, opts).then((response) => {
+      if (response.status == 404) {
+        this.thingsElement.innerHTML = this.THING_NOT_FOUND_MESSAGE;
+        return;
+      }
+
+      response.json().then((description) => {
```

Comment:
nit: this could be
```
  if (response.status == 404) {
    return Promise.resolve();
  }

  return response.json();
}).then((description) => {
```
to remove an unnecessary level of indentation and the duplication of `this.thingsElement.innerHTML = this.THING_NOT_FOUND_MESSAGE;`